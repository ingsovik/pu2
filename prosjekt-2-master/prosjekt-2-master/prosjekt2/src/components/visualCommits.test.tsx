import * as React from 'react';
import renderer from 'react-test-renderer'
import CommitsGraph, {dateBool, getDaysInMonth} from "./visualCommits";

test('Snapshot commits', () =>{
    const tree = renderer.create(<CommitsGraph />)
    expect(tree).toMatchSnapshot()

})

test('SEP, Find number of days in chosen month', () => {
    const sept = new Date('2021-09-14');
    const resultDays = getDaysInMonth(sept);
    expect(resultDays).toBe(30)
})
test('JAN, Find number of days in chosen month', () => {
    const jan = new Date('2021-01-14');
    const resultDays = getDaysInMonth(jan);
    expect(resultDays).toBe(31)
})

test('Loaded, test getting date (day only) from object', ()=> {
    const serviceStatus = 'loaded';
    const dateObj = new Date('2021-09-07');
    const resultString = dateBool(dateObj, serviceStatus)
    expect(resultString).toBe('07')
})
test('Loaded and Null, test getting date (day only) from object', ()=> {
    const serviceStatus = 'loaded';
    const dateObj = null;
    const resultString = dateBool(dateObj, serviceStatus)
    expect(resultString).toBe('--')
})

test('Not loaded, test getting date (day only) from object', ()=> {
    const serviceStatus = 'error';
    const dateObj = {date: '2021-09-21'};
    const resultString = dateBool(dateObj, serviceStatus)
    expect(resultString).toBe(undefined)
})

