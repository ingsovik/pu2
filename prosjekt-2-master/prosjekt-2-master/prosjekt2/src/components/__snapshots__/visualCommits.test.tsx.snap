// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot commits 1`] = `<div />`;
